### PR TITLE
Separated main.tf into smaller files and used variables instead of ha…

### PR DESCRIPTION
**Pull request by chubbymilo**
- Re-structured the project
- Removed hardcoded values, and used variable instead

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
+variable "aws_region" {
```
Comment (chubbymilo) [inline variables.tf:1]: I should have added descriptions and type. will add them

File: provider.tf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+provider "aws" {
+  region = var.aws_region
+}
+
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "5.50.0"
```
Comment (chubbymilo) [inline provider.tf:9]: Updated to the latest version to use Python 3.11 run time.

File: variables.tf (hunk 1 above)
Comment (ryan-caarter) [reply variables.tf:1]: Try adding validation rules 😜

File: variables.tf [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  type        = string
+  description = "The name of the lambda function from the Python file"
+}
+
+variable "lambda_run_time" {
+  default     = "python3.11"
+  type        = string
+  description = "The run time of the lambda function"
+  validation {
+    condition     = can(regex("python3.1\\d", var.lambda_run_time))
```
Comment (ryan-caarter) [inline variables.tf:34]: This is awesome

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+variable "lambda_permission_name" {
+  default     = "lambda-permissions"
+  type        = string
+  description = "The name of the ami role policy for accessing the lambda"
+}
+
+variable "file_type" {
+  default     = "zip"
```
Comment (ryan-caarter) [inline variables.tf:52]: This and output path don't really need to be variables as they don't change

Comment (ryan-caarter) [reply variables.tf:52]: And they won't change in a module either

Comment (chubbymilo) [reply variables.tf:52]: True true! thanks! I will remove them in the next PR. will merge this one and create a new PR

**Pull request merged**